__tests__/http/auth-gateway-test.js

```diff
@@ -0,0 +1,93 @@
+/* global describe, it, beforeEach */
+'use strict';
+
+var proxyquire   = require('proxyquireify')(require);
+var sinon        = require('sinon');
+var expect       = require('chai').expect;
+var EventEmitter = require('events').EventEmitter;
+
+var TOKEN_URI    = 'TOKEN_URI';
+var HOSTNAME     = 'HOSTNAME';
+var ACCESS_TOKEN = 'ACCESS_TOKEN';
+
+describe('auth-gateway', function () {
+    var HttpGateway, AuthGateway, authGateway, https;
+
+    beforeEach(function () {
+        https = sinon.spy();
+
+        HttpGateway = proxyquire(
```

Comment:
Proxyquire doesn't work with webpack. You've got to use https://github.com/jhnns/rewire-webpack